python/google/protobuf/internal/decoder.py

```diff
@@ -85,18 +85,17 @@
 __author__ = 'kenton@google.com (Kenton Varda)'
 
 import struct
-import sys  ##PY25
-_PY2 = sys.version_info[0] < 3  ##PY25
 from google.protobuf.internal import encoder
 from google.protobuf.internal import wire_format
+from google.protobuf.internal.utils import bytestr_to_string, b
 from google.protobuf import message
 
 
 # This will overflow and thus become IEEE-754 "infinity".  We would use
 # "float('inf')" but it doesn't work on Windows pre-Python-2.6.
-_POS_INF = 1e10000
-_NEG_INF = -_POS_INF
-_NAN = _POS_INF * 0
+_POS_INF = float('inf')
+_NEG_INF = float('-inf')
+_NAN = float('nan')
```

Comment:
I'm PyCharm IDE lead and dev and just have checked the usage statistics for the 2014. Python 2.5 and Python 2.4 use 1.41% and 0.16% of PyCharm users respectively. So we probably will abandon supporting 2.4 soon, but 2.5 should be still supported. If protobuf will drop support for 2.5 we'll have to bundle 2 different versions and I doubt that it would be easy to match them with 2 different versions on Java side. So it could be quite a pain.